tools/dissector/sample_manager.cpp

```diff
@@ -251,6 +261,19 @@ namespace OpenDDS
 #endif
 
     //--------------------------------------------------------------------
+
+    Sample_Manager::~Sample_Manager()
+    {
+      if (hf_array_ != NULL) {
+        delete [] hf_array_;
+      }
+      // Free Dynamically Generated Field Names
+      while (!field_names_.empty()) {
+        free(field_names_.front());
```

Comment:
I think these were allocated with ACE_OS::strdup, so this should be ACE_OS::free to match.